commands/uploadphotos.js

```diff
@@ -0,0 +1,57 @@
+import { google } from "googleapis"
+import { authorize } from "../utilities/google-auth.js"
+import fs from "fs"
+import { ContextMenuCommandBuilder, ApplicationCommandType } from "discord.js"
+import axios from "axios"
+import path from "path"
+
+export const data = new ContextMenuCommandBuilder().setName("upload photos").setType(ApplicationCommandType.Message)
+
+export async function execute(interaction) {
+  if (!interaction.isMessageContextMenuCommand()) return
+  const message = await interaction.channel.messages.fetch(interaction.targetId)
+  await interaction.deferReply() // Discord requires an acknowledgement within 3 seconds. this allows the response to be deferred, with an "<application> is thinking..." response in the meantime
+
+  let auth = await authorize()
+  const drive = google.drive({ version: "v3", auth })
+  const folderId = "[FOLDER ID]" // replace this with the id of the desired folder (the part of the url after "folders/")
+  const downloadDir = path.join(import.meta.dirname, "..", "downloads")
```

Comment:
Since we don't want to save the files, we should probably save the files in a temporary directory, which can be created using `tmpdir()` + `mkdtemp()`

Reply:
Alternatively, you can look at streaming the data directly, which is probably the more ideal solution though it'll be a bit harder to get working - create a file read stream pointing to the remote attachment download url and stream it to Google Drive. Something like https://stackoverflow.com/questions/65976322/how-to-create-a-readable-stream-from-a-remote-url-in-nodejs would probably work - you might have to play around a bit.